geth/jail/jail_test.go

```diff
@@ -195,6 +197,34 @@ func (s *JailTestSuite) TestJailFetch() {
 	<-wait
 }
 
+func (s *JailTestSuite) TestJailRPCAsyncSend() {
+	require := s.Require()
+	require.NotNil(s.jail)
+
+	s.StartTestNode(params.RopstenNetworkID)
+	defer s.StopTestNode()
+
+	// load Status JS and add test command to it
+	s.jail.BaseJS(baseStatusJSCode)
+	s.jail.Parse(testChatID, rxSendJS)
+
+	vm, err := s.jail.JailCellVM(testChatID)
+	require.NoError(err)
+	require.NotNil(vm)
+
+	// internally (since we replaced `web3.send` with `jail.Send`)
+	// all requests to web3 are forwarded to `jail.Send`
+	_, err = vm.Run(`
+		_status_catalog.commands.sendAsync({
+			"from": "` + TestConfig.Account1.Address + `",
+			"to": "0xf82da7547534045b4e00442bc89e16186cf8c272",
+			"value": "0.000001"
+		})
+	`)
```

Comment:
As the new async call is made in a goroutine with possible parallelism, it'd be wise to write a test which calls sendAsync simultaneously, say, in 10 goroutines to ensure there's no panic.
Anyway, please, wait until I merge `whisper-on-geth1.6.1` in `develop` before doing it as some synchronization is already done there.